projects/arlas-toolkit/src/lib/components/config-manager/config-menu/config-menu.component.ts

```diff
@@ -125,11 +135,44 @@ export class ConfigMenuComponent implements OnInit {
       }
       case ConfigActionEnum.DUPLICATE:
       case ConfigActionEnum.RENAME:
-      case ConfigActionEnum.SHARE: {
         if (action.config && action.config.id) {
           this.getDialogRef(action).subscribe(() => this.actionExecutedEmitter.next(action));
         }
         break;
+      case ConfigActionEnum.SHARE: {
+        if (action.config && action.config.id &&
+          action.config.value &&
+          JSON.parse(action.config.value).arlas &&
+          JSON.parse(action.config.value).arlas.server &&
+          JSON.parse(action.config.value).arlas.server.url) {
+          const arlasServerUrl = JSON.parse(action.config.value).arlas.server.url;
+          const configuration: Configuration = new Configuration();
+          const arlasExploreApi: ExploreApi = new ExploreApi(
+            configuration,
+            arlasServerUrl,
+            window.fetch
+          );
+          let headers = {};
+          if (this.arlasSettings.getAuthentSettings() && this.arlasSettings.getAuthentSettings().auth_mode === 'iam') {
```

Comment:
You should declare a cons settings before this if/else to avoid calling the method 4 times